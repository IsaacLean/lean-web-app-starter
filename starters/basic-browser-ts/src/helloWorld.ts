/**
 * Create a div element with the contents "Hello World!"
 */
export default function helloWorld() {
    const helloWorldDiv = document.createElement('div')
    helloWorldDiv.textContent = 'Hello World!'

    return helloWorldDiv
}
