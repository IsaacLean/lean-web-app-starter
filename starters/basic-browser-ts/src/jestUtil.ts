/**
 * Clear the root element.
 * This is intended to completely empty the root element's children between tests
 * so test results can be consistent.
 * @param rootEl Optionally pass in the rootEl to be cleared, otherwise a query for the element with the root ID will be made
 */
export const clearRootEl = (rootEl?: HTMLElement) => {
    const r = rootEl ? rootEl : document.getElementById('root')

    if (!r) {
        throw new Error('HTML element with an ID of "root" was not found.')
    }

    r.innerHTML = ''
}

/**
 * Setup the root element.
 * The root div element is intended for other elements to mount to for testing.
 * @returns The new root element
 */
export const setupRootEl = () => {
    const rootEl = document.createElement('div')
    rootEl.setAttribute('id', 'root')
    document.body.replaceChildren(rootEl)

    return rootEl
}
