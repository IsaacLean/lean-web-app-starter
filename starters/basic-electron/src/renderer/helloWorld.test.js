import helloWorld from './helloWorld'
import { clearRootEl, setupRootEl } from './jestUtil'

beforeAll(() => setupRootEl())

afterEach(() => clearRootEl())

test('helloWorld function renders "Hello World!" div element', () => {
    expect(helloWorld()).toMatchSnapshot()
})
