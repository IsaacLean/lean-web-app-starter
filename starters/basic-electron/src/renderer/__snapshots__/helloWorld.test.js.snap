// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`helloWorld function renders "Hello World!" div element 1`] = `
<div>
  Hello World!
</div>
`;
